Enable Azure CI with custom images for proprietary code

Lnaden opened the pull request:
This PR creates an Azure Pipeline which uses a custom Docker image we (MolSSI)
maintain and have exclusive access to through Azure Container Registry.

On this image we can house non-public code which will enable us to test
Engine through integrated CI on these codes securely.

I will say, there are lots of shenanigans and chain of about 20-25 different
commits of trying different things which got squished into this, but I can
confirm it works. Once this is merged in, I will activate it on the
Azure side for Engine itself and we can see it in action.

Right now, the image only contains a static MOPAC binary, but we could expand it
to include packages, or make other images and set up a strategy of different
containers to run through the pipeline. Thanks to Jonathan Moussa (@godotalgorithm)
for helping get the binary.

Here is what one of those runs looks like btw: https://dev.azure.com/MolSSI/QCArchive/_build/results?buildId=41

loriab commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,53 @@
+  - script: |
+      source $(conda_source)
+      conda activate test
+      pytest -v .
```
could add a `-rs` to get a summary of why stuff skipped

dgasmith commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,53 @@
+  containers:
+    -   container: 'qcengineci' # Arbitrary name for reference in pipelines
+        image: qcarchive.azurecr.io/qcengineci:latest # pointer to the Image in our Azure CR
+        endpoint: "QCArchive Azure Container Registry" # Name of the Service Connection the pipeline is configured with
```
Can you describe a bit about how the authentication is handled? Is this a private Azure pw?

Lnaden commented:
> this looks too simple for the power of what it does. do other things need to go in a README for documentation? 
> Can you describe a bit about how the authentication is handled? Is this a private Azure pw?

The YAML side of things is the simple part of this, I am working on a full write up for how everything else is handled as a markdown file. Stay tuned.

Lnaden commented:
I have renamed the YAML file to be a hidden file, and I have added in a full "How to set this all up from near-scratch" markdown file in the devtools folder:

https://github.com/MolSSI/QCEngine/blob/58088c34459cf96ea4dcd26b91f996897a86e306/devtools/azure_ci_images.md

If this addresses the concerns about security and how-tos, then I think we can merge this. If things are still unclear though, let me know and I will keep writting!

loriab commented on devtools/azure_ci_images.md:
```diff
@@ -0,0 +1,231 @@
+
+### Linking the services
+
+1. Sign into the account which meets the pre-requisites through the DevOps login.
```
`0.`

Lnaden replied:
Markdown doesn't care about numbered list ordering. I could have just put `1. 1. 1. 1111. 1. 11.` and it would have ordered them correctly in render. It does look a bit odd to the author though. I fixed it anyways.

loriab commented on devtools/azure_ci_images.md:
```diff
@@ -0,0 +1,231 @@
+  
+### Creating the Image
+
+1. Start a new [Dockerfile](https://docs.docker.com/engine/reference/builder/) with the following one lline header:
```
lline

loriab commented on devtools/azure_ci_images.md:
```diff
@@ -0,0 +1,231 @@
+
+### Running against multiple images
+
+One avenue for future iomprovements is having multiple images on the ACR we want to reference. For that, we can
```
improvements

Lnaden commented:
Thanks for the reviews. I'll merge this when it all passes and then point the CI here at engine.
